include/smp/lock.h

```diff
@@ -22,86 +22,48 @@
 typedef enum {
     CLHState_Granted = 0,
     CLHState_Pending
-} clh_qnode_state_t;
-
-typedef struct clh_qnode {
-    clh_qnode_state_t value;
-
-    PAD_TO_NEXT_CACHE_LN(sizeof(clh_qnode_state_t));
-} clh_qnode_t;
-
-typedef struct clh_qnode_p {
-    clh_qnode_t *node;
-    clh_qnode_t *next;
-    /* This is the software IPI flag */
+} clh_req_state_t;
+
+/* Lock request */
+typedef struct clh_req {
+    clh_req_state_t state;
+} ALIGN(L1_CACHE_LINE_SIZE) clh_req_t;
```

Comment:
It's not necessary, but it matches what the paper does. Having it as a structure makes it clearer that it's an object that's passed around, so I think I prefer it as it is. It's a very clever algorithm and making it as easy to understand as possible is the goal here.

To me it would be more logical to have the update of `node->myreq` together with the `node->myreq->state = CLHState_Pending`, but the paper also does it this way.